backend/dgraph/src/drug_interaction/insert_interaction.rs

```diff
@@ -0,0 +1,162 @@
+    use crate::insert_interaction_group::InteractionGroupInput;
+    use crate::InteractionGroupRef;
+    use util::uuid::uuid;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_insert_interaction() {
+        let client = DgraphClient::new("http://localhost:8080/graphql");
+
+        let interaction_id = uuid();
+        let interaction_group1_id = uuid();
+        let interaction_group2_id = uuid();
+
+        let interaction_name = "NSAIDs/ACE Inhibitors".to_string();
+
+        // Create the first Interaction Group
+        let interaction_group = InteractionGroupInput {
+            interaction_group_id: interaction_group1_id.clone(),
+            name: "NSAIDs".to_string(),
+            description: Some("Non-steroidal anti-inflammatory drugs".to_string()),
+            drugs: vec![DrugCode {
+                code: "294d8414".to_string(),
+            }],
+        };
+
+        let result = insert_interaction_group(&client, interaction_group, true).await;
+        if result.is_err() {
+            println!(
+                "insert_interaction_group err: {:#?} {:#?}",
+                result,
+                result.clone().unwrap_err().json()
+            );
+        }
+        assert!(result.is_ok());
+
+        // Create the second Interaction Group
+        let interaction_group = InteractionGroupInput {
+            interaction_group_id: interaction_group2_id.clone(),
+            name: "ACE Inhibitors".to_string(),
+            description: Some("Angiotensin-converting enzyme inhibitors".to_string()),
+            drugs: vec![DrugCode {
+                code: "ac07394b".to_string(),
+            }],
+        };
+
+        let result = insert_interaction_group(&client, interaction_group, true).await;
+        if result.is_err() {
+            println!(
+                "insert_interaction_group err: {:#?} {:#?}",
+                result,
+                result.clone().unwrap_err().json()
+            );
+        }
+        assert!(result.is_ok());
+
+        // Create a new Interaction
+
+        let interaction = DrugInteractionInput {
+            interaction_id: interaction_id.clone(),
+            name: interaction_name.clone(),
+            severity: DrugInteractionSeverity::Severe,
+            description: "NSAIDs may diminish the antihypertensive effect of ACE Inhibitors."
+                .to_string(),
+            action: "Monitor therapy".to_string(),
+            reference:
+                "https://www.drugs.com/drug-interactions/ace-inhibitors-with-nsaids-11-0-0-940-0"
+                    .to_string(),
+            drugs: vec![DrugCode {
+                code: "294d8414".to_string(),
+            }],
+            groups: vec![
+                InteractionGroupRef {
+                    interaction_group_id: interaction_group1_id.clone(),
+                },
+                InteractionGroupRef {
+                    interaction_group_id: interaction_group2_id.clone(),
+                },
```

Comment:
I mean I guess seeing they're stored as arrays this is possible at the dgraph level... but having the 1 drug and 2 groups is a bit confusing - only 2 things should be interacting right?

Reply:
Agreed, I probably just did this to see if it would work on not then forgot!

Reply:
[c022946](https://github.com/msupply-foundation/unified-codes/pull/574/commits/c0229460e4457ae1223ce0b01aac561145c4391c)